jdaviz/app.py

```diff
@@ -1124,18 +1103,26 @@ def vue_data_item_selected(self, event):
         """
         viewer_id, item_id, checked = event['id'], event['item_id'], event['checked']
         viewer_item = self._viewer_item_by_id(viewer_id)
+        replace = event.get('replace', False)
 
         if viewer_item is None:
             raise ValueError(f'viewer {viewer_id} not found')
 
-        if checked:
+        if replace:
+            selected_items = [item_id]
+        elif checked:
             selected_items = [*viewer_item['selected_data_items'], item_id]
         else:
             selected_items = list(filter(
                 lambda id: id != item_id, viewer_item['selected_data_items']))
 
         self._update_selected_data_items(viewer_id, selected_items)
 
+    def vue_data_item_remove(self, event):
+        """
+        """
+        self.data_collection.remove(self.data_collection[event['item_name']])
```

Comment:
Do we have to relink the data if the item being removed is the reference data?

Reply:
Yeah I've noticed other weird behavior in this PR with having multiple data selected in one viewer. My guess is that anything that touches linking (or removes links) may be the culprit.

Reply:
To summarize offline discussion: this method is when clicking the delete (not uncheck) button in the data menu so should completely and permanently remove the item app-wide.

Reply:
And there is no check on whether this item is reference data or just fitted model. Isn't that a little dangerous or is this by design?

Reply:
Right, there aren't any checks.  Since this is a `vue_` method, I think that is probably fine (this should never be called by a user).  But that does take us back to the discussion in https://github.com/spacetelescope/jdaviz/pull/1313#discussion_r871519596 - the best way to forbid deleting entries that shouldn't be deleted is by simply not showing a delete button on those entries... so maybe for now we should extend that to _only_ allow deleting entries that were products of plugins?

Reply:
> so maybe for now we should extend that to only allow deleting entries that were products of plugins?

Is this easier to implement? If so, might be good enough for now, and we can worry about deleting loaded data later.